lib/services/elasticsearch.js

```diff
@@ -152,1421 +163,1382 @@ class ElasticSearch extends Service {
-          this.throw('document_not_found', esRequest.id);
-        }
-        // injecting retryOnConflict default configuration
-        if (
-          !_.has(esRequest, 'retryOnConflict')
-          && this.config.defaults.onUpdateConflictRetries > 0
-        ) {
-          esRequest.retryOnConflict = this.config.defaults.onUpdateConflictRetries;
-        }
+  deleteByQuery (
+    index,
+    collection,
+    query,
+    { refresh=false, from=null, size=null } = {})
+  {
+    const esRequest = {
+      index: this._getESIndex(index, collection),
+      body: { query },
+      scroll: '30s',
+      from,
+      size,
+      refresh
+    };
 
-        // Add metadata
-        esRequest.body = {
-          doc: {
-            _kuzzle_info: {
-              active: false,
-              deletedAt: Date.now(),
-              updater: this._getUserId(request)
-            }
-          }
-        };
+    if (! _.isPlainObject(query)) {
+      return this.reject('empty_query');
+    }
 
-        return this.client.update(esRequest)
-          .then(r => this.refreshIndexIfNeeded(esRequest, r))
-          .catch(error => Bluebird.reject(
-            this.esWrapper.formatESError(error))
-          );
-      })
-      .catch(error => Bluebird.reject(
-        this.esWrapper.formatESError(error))
-      );
+    return this.client.deleteByQuery(esRequest)
+      .then(({ body }) => ({
+        total: body.total,
+        deleted: body.deleted,
+        failures:
+          body.failures.map(({ shardId, reason }) => ({
+            shardId,
+            reason
+          }))
+      }))
+      .catch(error => this.esWrapper.reject(error));
   }
 
   /**
-   * Delete all documents matching the provided filters
+   * Creates a new index.
    *
-   * @param {Request} request
-   * @returns {Promise} resolve an object with ids
+   * This methods does not create an index because they don't exists physicaly
+   * inside Elasticsearch since each collection is an Elastic index.
+   * This methods resolves if the index name does not already exists either as
+   * internal or public index.
+   *
+   * @param {String} index - Index name
+   *
+   * @returns {Promise}
    */
-  deleteByQuery(request) {
-    const
-      esRequestSearch = initESRequest(request, ['from', 'size', 'scroll']),
-      esRequestBulk = initESRequest(request, ['refresh']),
-      ts = Date.now();
+  createIndex (index) {
+    return this.client.cat.indices({ format: 'json' })
+      .then(({ body }) => {
+        const esIndexes = body.map(({ index: name }) => name);
 
-    assertWellFormedRefresh(esRequestBulk);
+        for (const esIndex of esIndexes) {
+          const indexName = this._extractIndex(esIndex);
 
-    esRequestSearch.body = request.input.body;
-    esRequestSearch.scroll = '30s';
-    esRequestBulk.body = [];
+          if (index === indexName) {
+            const internalMessage = esIndex[0] === INTERNAL_PREFIX
+              ? ' Index is used internaly by Kuzzle.'
```

Comment:
We don't now for sure. 

```suggestion
              ? ' A private index of that name already exists.'
```

If you don't approve that suggestion, then you at least need to fix this typo:

```suggestion
              ? ' Index is used internally by Kuzzle.'
```